torch/storage.py

```diff
@@ -146,20 +101,91 @@ def share_memory_(self):
         return self
 
     @classmethod
-    def _new_shared(cls, size):
+    def _new_shared(cls, *, nbytes):
         """Creates a new storage in shared memory with the same data type"""
         from torch.multiprocessing import get_sharing_strategy
         if cls.is_cuda:
-            return cls(size)
+            return cls(nbytes)
         elif get_sharing_strategy() == 'file_system':
-            return cls._new_using_filename(size)
+            return cls._new_using_filename(nbytes)
         else:
-            return cls._new_using_fd(size)
+            return cls._new_using_fd(nbytes)
 
 
 def _load_from_bytes(b):
     return torch.load(io.BytesIO(b))
 
 
-_StorageBase.type = _type  # type: ignore[assignment]
 _StorageBase.cuda = _cuda  # type: ignore[assignment]
+
+
+def _dtype_to_pickle_storage_type_map():
+    return {
+        torch.double: 'DoubleStorage',
+        torch.float: 'FloatStorage',
+        torch.half: 'HalfStorage',
+        torch.long: 'LongStorage',
+        torch.int: 'IntStorage',
+        torch.short: 'ShortStorage',
+        torch.int8: 'CharStorage',
+        torch.uint8: 'ByteStorage',
+        torch.bool: 'BoolStorage',
+        torch.bfloat16: 'BFloat16Storage',
+        torch.cdouble: 'ComplexDoubleStorage',
+        torch.cfloat: 'ComplexFloatStorage',
+        torch.qint8: 'QInt8Storage',
+        torch.qint32: 'QInt32Storage',
+        torch.quint8: 'QUInt8Storage',
+        torch.quint4x2: 'QUInt4x2Storage',
+    }
+
+def _pickle_storage_type_to_dtype_map():
+    if not hasattr(_pickle_storage_type_to_dtype_map, 'cache'):
+        _pickle_storage_type_to_dtype_map.cache = {  # type: ignore[attr-defined]
+            val: key for key, val in _dtype_to_pickle_storage_type_map().items()}
+    return _pickle_storage_type_to_dtype_map.cache  # type: ignore[attr-defined]
```

Comment:
A more compact way to write this is just `@lru_cache`

Reply:
Oh cool, didn't know about that!